packages/modular-scripts/src/addPackage.ts

```diff
@@ -203,22 +206,24 @@ async function addPackage({
     );
   }
 
-  const packageTypePath = path.dirname(newModularPackageJsonPath);
-  // create a new package source folder
-  await fs.mkdirp(packagePath);
-  await fs.copy(packageTypePath, packagePath, {
+  const templatePath = path.dirname(templatePackageJsonPath);
+  // Create new package directory
+  await fs.mkdirp(newPackagePath);
+  // Copy files from template directory to new package directory
+  await fs.copy(templatePath, newPackagePath, {
     recursive: true,
     filter(src) {
       return !(path.basename(src) === 'package.json');
     },
   });
 
-  const packageFilePaths = getAllFiles(packagePath);
+  const packageFilePaths = getAllFiles(newPackagePath);
 
   // If we get our package locally we need to allowlist files like yarn publish does
+  // If package.json specifies "files" to include, filter, otherwise allow all files
   const packageAllowlist = globby
-    .sync(modularTemplatePackageJson.files || ['*'], {
-      cwd: packagePath,
+    .sync(modularTemplatePackageJson.files || ['**'], {
```

Comment:
Maybe a bit out of scope, but it turns out there's a npm package to do just this the "official" way, instead of the globby hack we're using here: https://github.com/npm/npm-packlist

Since we're improving local `files` filtering anyway, what do you think if we just use this package instead?

Reply:
Switched to packlist with my latest changes - surprised that it requires @npmcli/arborist and added over 50 indirect dependencies to Modular, seems excessive for such a small bit of functionality

Reply:
That is a bit suprising. Taking a look, I guess it makes sense why: `Arborist` is npm's internal representation of `node_modules`. The [arborist library](https://github.com/npm/cli/tree/latest/workspaces/arborist) (TIL it exists) is a bit similar to our workspace resolver package (but different outcomes - eg building an ideal tree whereas we're interested in ancestors and other things). So when we add `npm-packlist` we get all of that as devDeps.

Assuming that we're working within the bounds of npm/yarn (yarn being npm-compatible) (not sure about pnpm but my assumption would be that it's also npm-compatible) for a long time (ignoring things like Deno and Bun that seem definitely out of scope), we get the guarantee of being spec/design compliant, less can go wrong. So there is some value for sure.

An example is the "common cruft files" described in that package's readme ([see rule number 5](https://github.com/npm/npm-packlist)) - I doubt `**` results in full/proper compliance vs using `npm-packlist`.

Reply:
Thanks, that's some valuable insight and some really good points. I see the value in using it but I'm seeing it generate some curious errors in the tests which I'll investigate tomorrow.